server/src/main/java/org/elasticsearch/action/admin/cluster/repositories/delete/TransportDeleteRepositoryAction.java

```diff
@@ -85,4 +85,16 @@ public Optional<String> reservedStateHandlerName() {
     public Set<String> modifiedKeys(DeleteRepositoryRequest request) {
         return Set.of(request.name());
     }
+
+    @Override
+    protected void validateForReservedState(DeleteRepositoryRequest request, ClusterState state) {
+        super.validateForReservedState(request, state);
+
+        validateForReservedState(
+            projectResolver.getProjectMetadata(state).reservedStateMetadata().values(),
+            reservedStateHandlerName().get(),
+            modifiedKeys(request),
+            request.toString()
+        );
+    }
```

Comment:
This is copied from the same check from other other reserved project-state actions such as [this one](https://github.com/elastic/elasticsearch/blob/77b459c45493e901d8898960515c8db533efcd81/server/src/main/java/org/elasticsearch/action/admin/indices/template/delete/TransportDeleteComposableIndexTemplateAction.java#L94-L103). There is room for de-duplication as suggested by the [comment](https://github.com/elastic/elasticsearch/blob/77b459c45493e901d8898960515c8db533efcd81/server/src/main/java/org/elasticsearch/action/admin/indices/template/put/TransportPutComponentTemplateAction.java#L126), but I'll leave it out from this PR.